packages/hydrogen-codegen/src/schema.ts

```diff
@@ -0,0 +1,15 @@
+// This comment is used during ESM build:
+//! import {createRequire} from 'module'; const require = createRequire(import.meta.url);
+export const getSchema = () =>
+  require.resolve('@shopify/hydrogen-react/storefront.schema.json');
+
+let staticSchema = '';
+
+try {
+  staticSchema = getSchema();
+} catch {
+  // This can happen at build time or when '@shopify/hydrogen-react' is not found.
+  // Generally this shouldn't be an issue in real apps so let's ignore the error.
```

Comment:
The thing is that this situation could happen when importing `{plugin}` without using Hydrogen itself and I thought it would be nice to support that situation (just for the sake of using the same codegen strategy we have here, maybe for other features for Remix).
But maybe it's OK to get a warning about it anyway, even if you're not installing Hydrogen on purpose.